### PR TITLE
Share slice of bytes

### PR DESCRIPTION
**Pull request by WaffleLapkin**
r? @Nilstrieb 
cc @noamtashma

### CODE REVIEW

File: compiler/rustc_data_structures/src/sync.rs [hunk 1]
```diff
@@ -52,6 +52,8 @@ pub use vec::{AppendOnlyIndexVec, AppendOnlyVec};
 
 mod vec;
 
+pub type MetadataRef = OwnedSlice;
```
Comment (Nilstrieb) [inline compiler/rustc_data_structures/src/sync.rs:55]: This type alias doesn't really belong here anymore when it's not cfged anymore.

Comment (WaffleLapkin) [reply compiler/rustc_data_structures/src/sync.rs:55]: Do you have an idea of where to put it? (do we even need it now?)

Comment (Nilstrieb) [reply compiler/rustc_data_structures/src/sync.rs:55]: Deleting it is probably the best idea

**Review by Noratrieb — APPROVED** [review]
r=me if perf is not sad, which it probably won't be

**Comment by noamtashma** [comment]
Thanks!

A few questions/nits: 
*    I wonder if this should be changed to `SharedSlice` now, but I guess it should really be called `SharedOwnedSlice` or `LrcSlice`, which imo just sound bad.
*    A few comments still refer to `box`.
*    Shouldn't the methods in `MetadataLoader` use `MetadataBlob` just so that the types would be a bit more clear?

**Review by Noratrieb — APPROVED** [review]

**Pull request merged**